R/pkg/R/DataFrame.R

```diff
@@ -594,6 +594,35 @@ setMethod("repartition",
             dataFrame(sdf)
           })
 
+#' RepartitionByColumn
+#'
+#' Return a new SparkDataFrame which has as many partitions as the number of unique
+#' groups identified by column(s) values which are being specified by the input.
+#'
+#' @param x A SparkDataFrame
+#' @param col The column by which the partitioning will be performed
+#'
+#' @family SparkDataFrame functions
+#' @rdname repartition
+#' @name repartition
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlContext <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- read.json(sqlContext, path)
+#' newDF <- repartitionByColumn(df, df$col1, df$col2)
+#'}
+setMethod("repartitionByColumn",
```

Comment:
I think @felixcheung's proposal is good - better to not introduce a new keyword if the existing one suffices